### PR TITLE
Bug -  4761 -  Skip add skill links

### LINKED ISSUE #4761: Skills Picker- Keyboard / Focus Trap
## Description
When the user tries to navigate the skills picker (or any scrollable element) they are confronted with a keyboard trap that keeps them in the element.  Any element should be navigable by keyboard using the following keys: 
1.  Tab into component,
2. Use arrow keys to navigate through links↑/↓ or ←/→ - increase or decrease slider value, Home/End - beginning or end, 
3. **Tab out of component**

This fix will be important for Fable Testing in the coming weeks :).

## Screenshot: Production
[Focus Trap issue - Skills picker](https://user-images.githubusercontent.com/101275838/202018126-3d926d0b-2915-4646-9c72-746cb9d1ca16.PNG)

### PR DESCRIPTION
**Pull request by esizer**
## 👋 Introduction

This adds a skip link to allow users to skip the long list of skills in the skill picker.

## 🧪 Testing

> **Note**
> These tests should be completed with a screen reader on and keyboard only

1. Build talentsearch or admin
2. Navigate to a skill picker
3. Tab past the search bar
4. Ensure you hear the skip link text
5. Ensure interacting with the link takes you past the skill list

## 🤖 Robot Stuff

Resolves #4761

### CODE REVIEW

**Comment by MaddyDaigle** [comment]
Just realized I never sent the explainer text!  Sorry about that. 

Here is what I think it should say. @esizer & @RM-1978 - thoughts?

"Press ctrl + shift + esc while navigating through the list of skills to skip directly to the candidate results."

I think if they use the skip control it should bring them directly to the h3 heading `Results: # matching candidates` since that's the next logical heading/ section of the page.

**Comment by esizer** [comment]
> "Press ctrl + shift + esc while navigating through the list of skills to skip directly to the candidate results."

We might need to tweak the end of this to be more generic. We use the skill picker in different areas so you aren't always skipping to candidate results.

File: frontend/common/src/components/SkillPicker/SkillPicker.tsx [hunk 1]
```diff
@@ -168,17 +194,54 @@ const SkillPicker = ({
...
+      {isInSkillList ? (
+        <p
+          aria-live="polite"
+          data-h2-visibility="base(invisible)"
+          id="skill-list-help"
+        >
+          {intl.formatMessage({
+            id: "tRYI0f",
+            defaultMessage:
+              "Press ctrl + shift + esc while navigating through the list of skills to skip directly to your chosen skills.",
```
Comment (petertgiles) [inline frontend/common/src/components/SkillPicker/SkillPicker.tsx:222]: Mac users living in a Windows world will proabably know that ctrl -> command for them.  But does cmd+shift+esc work here too?

**Review by petertgiles — APPROVED** [review]
Works well for me!  Please don't merge quite yet.  I'd to see if Maddy can review first, too.

File: frontend/common/src/components/SkillPicker/SkillPicker.tsx (hunk 1 above)
Comment (esizer) [reply frontend/common/src/components/SkillPicker/SkillPicker.tsx:222]: It does not. However, we are working on the actual shortcut. @RM-1978 did some testing on a variety of screen readers last night and it turns out this interferes with a windows specific shortcut 😢

**Comment by MaddyDaigle** [comment]
As discussed with @esizer, here is the new accessibile functionality:

New text for screen reader: “Press ctrl + shift + esc while navigating through the list of skills to skip directly to your chosen skills.”

IF skip link (or skip keyboard shortcut) is triggered, bring user to selected skills list.  Add a header that is only shown to the screen reader to orient the user.

Header (only read to screen reader): “Selected skills”
If there are no selected skills (screen reader only):
“You have not selected any skills”.

**Comment by petertgiles** [comment]
Alt + Shift + Esc is already bound to something for me so I can't test right now  :cry: I doubt many of our users are on Linux so I wouldn't worry too much about me.

Is the use of a lot of punctuation (parenthesis and plus) in a message for screen readers problematic?

**Comment by esizer** [comment]
> Is the use of a lot of punctuation (parenthesis and plus) in a message for screen readers problematic?

It depends, for this specific string;

- `+` is read as plus or ignored
- `()` is read as left/right paren or ignored

The worst case scenario is that it is read as:

> "Press alt left parent option on a mac right paren plus shift plus escape..."

A good place to see is [Deuque's article on screen reader punctuation](https://www.deque.com/blog/dont-screen-readers-read-whats-screen-part-1-punctuation-typographic-symbols/).

**Comment by brindasasi** [comment]
am not forgetting this. my local is broken currently with nginx. setting up

**Comment by brindasasi** [comment]
> #4761

- [ ] Ensure you hear the skip link text 
- [ ] Ensure interacting with the link takes you past the skill list 

- [ ] Press the shortcut key (alt+shift+esc)
- [ ] Continue with the text and ensure you are past the skill list

**Comment by esizer** [comment]
> Screen reader didn't read the above text on mac. Tried many times.

I just tested (MacOS + VoiceOver) and the text is being read out. You need to manually move to the text to have it read out since VoiceOver only reads one block at a time.

1. Move focus to the list of skills with `tab`
2. Press the shortcut `option` + `shift` + `esc`
    - Focus moves to heading "Selected skills"
3. Move cursor using VoiceOver `control` + `option` + `right arrow`

**Review by brindasasi — APPROVED** [review]

**Review by yonikid15 — APPROVED** [review]
Eric skip link is working on my end! Nice work 👍

**Pull request merged**